pact/provider/core/src/test/java/org/arquillian/algeron/pact/provider/core/InteractionRunnerTest.java

```diff
@@ -129,11 +160,38 @@ public void should_throw_exception_when_state_param_is_not_empy_nor_map() {
 
     @Provider("planets_provider")
     @ContractsFolder("pacts")
-    public static class PactProviderWithWrongStateMethod {
+    public static class PactProviderWithIntegerParameterStateMethod {
+
+        private int numberOfCakes = 0;
+
+        @State("I have (\\d+) cukes in my belly")
+        public void stateMethod(int numberOfCakes) {
+            this.numberOfCakes = numberOfCakes;
```

Comment:
Ok, now I'm confused :D Are we talking about cukes (cucumbers) or cakes?

![will_smith_hmm](https://cloud.githubusercontent.com/assets/719616/20661186/bcc8c2fe-b54d-11e6-9a59-ce5891f1d124.gif)